recipes/folly/all/patches/2023-001-compiler-flags.patch

```diff
@@ -0,0 +1,19 @@
+--- CMake/FollyCompilerUnix.cmake
++++ CMake/FollyCompilerUnix.cmake
+@@ -12,7 +12,7 @@
+ # See the License for the specific language governing permissions and
+ # limitations under the License.
+ 
+-set(CMAKE_CXX_FLAGS_COMMON "-g -Wall -Wextra")
```

Comment:
I appreciate that this patch was already there - but it's a good opportunity to check why it's here in the first place. There's nothing wrong with `-g` - it's still optimized binaries in Release mode. Either way, not related to this PR if this was already there.

Reply:
My guess it's that is an attempt at producing a pure `Release` package without debug symbols. We can consider this a concious decision by Folly and remove the pactch if you want.

Reply:
I'd say it disrespects the user's expectations for `Release`, since `-O` with `-g` is what `RelWithDebInfo` build type is for. On the other hand, it would be better addressed upstream rather than on CCI, so I'm in favor of dropping this patch.